bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -458,7 +459,14 @@ pub mod pallet {
 				Self::deposit_event(Event::SyncCommitteeUpdated {
 					period: update_finalized_period,
 				});
-			};
+			} else {
+				// If there was no sync committee in the update, but the finalized header provided
+				// is in the new period, we need to move the sync committee period forward.
+				if update_finalized_period == store_period + 1 {
+					ensure!(<NextSyncCommittee<T>>::exists(), <Error<T>>::NextSyncCommitteeUnknown);
```

Comment:
@yrong @vgeddes I added this check. One thing I dislike about the Ethereum consensus spec is that the separation between `verify_update` and `apply_update` makes it difficult to see what conditions are possible in `apply_update` and what has already been checked in `verify_update`.

I don't think it is possible for <NextSyncCommittee<T>> to be empty here because the header validation had to pass [BLS signature verification here](https://github.com/Snowfork/polkadot-sdk/blob/snowbridge/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L404) to get to this point. I would rather err on the side of being too cautious though. Let me know your thoughts.

Reply:
So I would prefer to move all check logics to verification phase and apply_update with only actions required.

Reply:
Yeah that sounds good. So are we going to disallow updates in the next period without a sync committee update, like I illustrated here: https://github.com/Snowfork/polkadot-sdk/pull/145#discussion_r1601123132